Generator/Templates/Shared/helper_dll_import.sbntxt

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace {{ namespace }}
+{
+    internal static class DllImport
+    {
+        #region Fields
+        
+        private const string _windowsDllName = "{{ windows_dll }}";
+        private const string _linuxDllName = "{{ linux_dll }}";
+        private const string _osxDllName = "{{ osx_dll }}";
+
+        private static readonly string DllName;
+
+        #endregion
+        
+        #region Constructors
+        
+        static DllImport()
+        {
+            DllName = GetDllImport();
+        }
+        
+        #endregion
+            
+        #region Methods
+        
+        public static void Initialize()
+        {
+            NativeLibrary.SetDllImportResolver(typeof(DllImport).Assembly, ImportResolver);
+        }
+        
+        private static IntPtr ImportResolver(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            if (NativeLibrary.TryLoad(DllName, assembly, DllImportSearchPath.System32, out IntPtr libHandle))
```

Comment:
This was not working for me until I've changed the `DllImportSearchPath` to `LegacyBehavior` (using Windows)

Reply:
Hm `LegacyBehavior` is working for me, too. I updated the branch to use this value.

@cameronwhite is it working for you, too?

Reply:
Yes, it's still working for me on macOS with the latest changes in this branch